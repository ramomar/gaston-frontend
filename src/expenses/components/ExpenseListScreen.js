import React from 'react';
import PropTypes from 'prop-types';
import Shapes from '../shapes'
import { Text } from 'grommet';
import { Screen, ScreenHeader, ScreenBody } from '../../foundation/components/screen';
import ExpenseList from './ExpenseList';

function ExpenseListScreen(props) {
  return (
    <Screen>
      <ScreenHeader
        center={<Text weight='bold' size='large'>Revisi√≥n de gastos</Text>} />
      <ScreenBody>
        <ExpenseList
          expenseGroups={props.expenseGroups}
          isFetching={props.isFetching}
          hasMore={props.hasMore}
          error={props.error}
          moreExpenses={props.moreExpenses}
          toExpenseReviewScreen={props.toExpenseReviewScreen} />
      </ScreenBody>
    </Screen>
  );
}

ExpenseListScreen.propTypes = {
  expenseGroups: PropTypes.arrayOf(Shapes.expenseGroup).isRequired,
  isFetching: PropTypes.bool.isRequired,
  hasMore: PropTypes.bool.isRequired,
  error: PropTypes.string,
  moreExpenses: PropTypes.func.isRequired,
  toExpenseReviewScreen: PropTypes.func.isRequired
};

export default ExpenseListScreen;
