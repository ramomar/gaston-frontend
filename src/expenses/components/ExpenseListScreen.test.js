import React from 'react';
import renderer from 'react-test-renderer';
import ExpenseListScreen from './ExpenseListScreen';
import { DateTime } from 'luxon';

it('renders correctly', () => {
  const expense = {
    'id': '0007182d-54cb-42b7-88fc-bbaba51db198',
    'amount': 150,
    'date': DateTime.fromISO('2017-03-19T05:29:02.700Z'),
    'note': 'Cena'
  };

  const expenseGroups = [
    {
      day: DateTime.fromISO('2017-03-19T05:29:02.700Z').startOf('day'),
      expenses: [expense]
    }
  ];

  const component =
    <ExpenseListScreen
      expenseGroups={expenseGroups}
      isFetching={true}
      hasMore={false}
      moreExpenses={() => null}
      toExpenseReviewScreen={() => null} />;

  const tree = renderer.create(component);

  expect(tree).toMatchSnapshot();
});
