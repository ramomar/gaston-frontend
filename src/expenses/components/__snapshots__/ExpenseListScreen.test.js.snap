// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 koruWX"
>
  <div
    className="StyledBox-sc-13pk1d4-0 fpgWwd"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 koruWX"
    />
    <div
      className="StyledBox-sc-13pk1d4-0 koruWX"
    >
      <span
        className="StyledText-sc-1sadyjn-0 dFefMJ"
        size="large"
      >
        Revisi√≥n de gastos
      </span>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 koruWX"
    />
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 koruWX"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 hCTAHO"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 ieLbCm"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 jNYyVh"
          style={
            Object {
              "position": "sticky",
              "top": 0,
            }
          }
        >
          <span
            className="StyledText-sc-1sadyjn-0 bnqisW"
          >
            18 de March
          </span>
          <div
            className="StyledBox-sc-13pk1d4-0 dRsJGz"
          >
            <span
              className="StyledText-sc-1sadyjn-0 hLzlgx"
              size="small"
            >
              -$
              150.00
            </span>
          </div>
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 vtIgb"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          <div
            className="StyledBox-sc-13pk1d4-0 cHMbYi"
          >
            <span
              className="StyledText-sc-1sadyjn-0 jOAXsc"
            >
              Cena
            </span>
          </div>
          <div
            className="StyledBox-sc-13pk1d4-0 gIKgZq"
          >
            <span
              className="StyledText-sc-1sadyjn-0 isVaoK"
              size="small"
            >
              -$
              150.00
            </span>
            <span
              className="StyledText-sc-1sadyjn-0 gNZQxf"
              size="small"
            >
              23:29
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
